aoc/src/main/scala/aoc2022/exercise02.scala

```diff
@@ -0,0 +1,71 @@
+package aoc2022
+
+import aoc2022.standardAoc.readInput
+
+import scala.io.Source
+
+
+
+object exercise02 extends App:
+
+  val transFunc = (action: String) =>  action match
+    case "A" => "Rock"
+    case "B" => "Paper"
+    case "C" => "Scissors"
+    case "X" => "Rock"
+    case "Y" => "Paper"
+    case "Z" => "Scissors"
+
+  val pointsChoice = (action: String) =>  action match
+    case "Rock"     => 1
+    case "Paper"    => 2
+    case "Scissors" => 3
+
+
+  val PointsOutcome = (action: (String,String)) =>  action match
+    case (x,y) if x == y         => 3
+    case ("Rock","Scissors")     => 6
+    case ("Paper", "Rock")       => 6
+    case ("Scissors", "Paper")   => 6
+    case _                       => 0
+
+  val actionForOutcome = (actionOutcome: (String,String)) =>  actionOutcome match
+    case (x,y) if y == "Y"     => x
+    case ("Rock","Z")          => "Paper"
+    case ("Rock","X")          => "Scissors"
+    case ("Scissors","Z")      => "Rock"
+    case ("Scissors","X")      => "Paper"
+    case ("Paper","Z")         => "Scissors"
+    case ("Paper","X")         => "Rock"
+
+
+  case class PlayGame(actionA: String, actionB: String,typeInput: String):
```

Comment:
im not fond of data switches - ie. a string as switch

try to pass a function value instead, eg.:

```
scala> def exponential(todo: List[Int], multiplier: Int = 1, postProcess: List[Int] => List[Int] = identity, accumulator: List[Int] = List.empty): List[Int] =
     |   if todo.isEmpty then postProcess(accumulator)
     |   else exponential(todo.tail, multiplier * 2, postProcess, accumulator :+ (todo.head * multiplier))
     | 
def exponential(todo: List[Int], multiplier: Int, postProcess: List[Int] => List[Int], accumulator: List[Int]): List[Int]
                                                                                                                                                                                                                                                                                                                                                       
scala> exponential(List(1,2,2,1,2,3,2,1,2,1,2))
val res0: List[Int] = List(1, 4, 8, 8, 32, 96, 128, 128, 512, 512, 2048)
```